waiter/test/waiter/scaling_test.clj

```diff
@@ -667,45 +667,95 @@
-                                  "app2" {:healthy-instances 5 :task-count 5 :expired-instances 0}
-                                  "app3" {:healthy-instances 0 :task-count 0 :expired-instances 0}
-                                  "app4" {:healthy-instances 15 :task-count 15 :expired-instances 0}}
+                                  "app4" {:target-instances 10}
+                                  "app5" {:target-instances 12}
+                                  "app6" {:target-instances 10}}
+                                 apply-scaling
+                                 5
+                                 test-scale-service
+                                 ;; service-id->router-state
+                                 {"app1" {:healthy-instances 5 :task-count 5 :expired-healthy-instances 0 :expired-unhealthy-instances 0}
+                                  "app2" {:healthy-instances 5 :task-count 5 :expired-healthy-instances 0 :expired-unhealthy-instances 0}
+                                  "app3" {:healthy-instances 0 :task-count 0 :expired-healthy-instances 0 :expired-unhealthy-instances 0}
+                                  "app4" {:healthy-instances 15 :task-count 15 :expired-healthy-instances 0 :expired-unhealthy-instances 0}
+                                  "app5" {:healthy-instances 10 :task-count 10 :expired-healthy-instances 7 :expired-unhealthy-instances 4}
+                                  "app6" {:healthy-instances 5 :task-count 5 :expired-healthy-instances 0 :expired-unhealthy-instances 0}}
+                                 ;; service-id->scheduler-state
                                  {"app1" {:instances 5 :task-count 5}
                                   "app2" {:instances 5 :task-count 5}
                                   "app3" {:instances 0 :task-count 0}
-                                  "app4" {:instances 15 :task-count 15}})]
-      (is (= 10 (get-in result ["app1" :target-instances])))
-      (is (= 5 (get-in result ["app2" :target-instances])))
-      (is (= 5 (get-in result ["app3" :target-instances])))
-      (is (= 10 (get-in result ["app4" :target-instances]))))))
+                                  "app4" {:instances 15 :task-count 15}
+                                  "app5" {:instances 10 :task-count 10}
+                                  "app6" {:instances 5 :task-count 5}}
+                                 max-expired-unhealthy-instances-to-consider)]
+      (clojure.pprint/pprint result)
+      (is (= {:target-instances 10, :scale-to-instances 10, :scale-amount 5} (get result "app1")))
+      (is (= {:target-instances 5, :scale-to-instances 5, :scale-amount 0} (get result "app2")))
+      (is (= {:target-instances 5, :scale-to-instances 5, :scale-amount 5} (get result "app3")))
+      (is (= {:target-instances 10, :scale-to-instances 10, :scale-amount -5} (get result "app4")))
+      (is (= {:target-instances 12, :scale-to-instances 21, :scale-amount 11} (get result "app5")))
```

Comment:
If we are limiting max-consider to 2, why are we scaling up by 11?

Reply:
I now understand why, but leaving this comment to illustrate the confusion that could have been avoided with comment.

Reply:
added comment